src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -354,18 +351,29 @@ bool reclaimOrphanedSession(FilePath* pSessionDir)
 
    BOOST_FOREACH(const FilePath& sessionDir, sessionDirs)
    {
+      // if the suspend file exists, this session is only sleeping, not dead
+      if (sessionSuspendFilePath(sessionDir).exists())
```

Comment:
Would it be worth the additional complexity to assume that 'stale' files here are considered as orphans? E.g. if the file modify time is more than a minute old?